### PR TITLE
ARC-63: Static analysis as rake

### PR DESCRIPTION
**Pull request by CalicoCoder**
Adding in the various gem files with bullet and jslint currently disabled.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -44,6 +44,14 @@ group :development do
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
+  # Brakeman is used to detect security risks in the application
+  gem 'brakeman', :require => false
+  # RuboCop is a static code analyzer tool that will ensure code quality
+  gem 'rubocop', require: false
+  # Looks for query optimations to avoid n+1 issues. Read more: https://sitepoint.com/silver-bullet-n1-problem
+  #gem 'bullet' # Has an issue with no support mongoid v6.0.0, re-asses later
```
Comment (mr2ert) [inline Gemfile]: Could you change this to:

```
# TODO: Has an issue with no support mongoid v6.0.0, re-asses later
#gem 'bullet'
```

To keep with the current TODO style?

File: config/environments/development.rb [hunk 2]
```diff
@@ -53,4 +53,10 @@
     config.web_console.whitelisted_ips = '0.0.0.0/0' # all ip-addresses
     config.logger = ActiveSupport::Logger.new(STDOUT) 
   end
+
+  #config.after_initialize do
```
Comment (mr2ert) [inline config/environments/development.rb:58]: Could you also add

```
# TODO: Has an issue with no support mongoid v6.0.0, re-asses later
```

Here too?

**Review by mr2ert — CHANGES_REQUESTED** [review]
Please update these

File: docker-compose.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version: '2'
+version: '2.0'
```
Comment (mr2ert) [inline docker-compose.yml]: Why did you need to make this change?  Was it not working before?

File: Rakefile [hunk 4]
```diff
@@ -2,5 +2,11 @@
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
 require_relative 'config/application'
+require 'rubocop/rake_task'
 
 Rails.application.load_tasks
+
+  desc 'Runs rubocop rake task'
+  task :static_analysis do
+    RuboCop::RakeTask.new
```
Comment (mr2ert) [inline Rakefile]: Can we run brakeman here as well? http://brakemanscanner.org/docs/rake/  We should update the `desc` as well if we do this.

File: docker-compose.yml (hunk 3 above)
Comment (CalicoCoder) [reply docker-compose.yml]: There was an error message previously, but it is no longer necessary.

File: Rakefile [hunk 5]
```diff
@@ -4,3 +4,13 @@
 require_relative 'config/application'
 
 Rails.application.load_tasks
+
+desc 'Runs rubocop and brakeman rake task'
+task :static_analysis, :output_files do |t, args|
```
Comment (mr2ert) [inline Rakefile]: Could we move this to `lib/tasks/static_analysis.rb`?

**Review by mr2ert — APPROVED** [review]

**Comment by mr2ert** [comment]
:shipit: Thanks for handling this.  Might want to add something to the README to say how to run this before you squash merge.

**Comment by CalicoCoder** [comment]
Updating README to include info about the newly added static analysis task

**Pull request merged**